File: johanna/bank.py
#!/usr/bin/env python3

from lyftplan import *

kb_tavling = KB("tävling", reps={1 : (110, "2017-11-11")})
bp_tavling = BP("tävling", reps={1 : (65, "2018-02-24")})
ml_tavling = ML("tävling", reps={1 : (130, "2018-01-20")})

"""
bp = BP("pekfinger, stopp", parent=bp_tavling, reps={1 : (65, "2018-01-20")}, time=5.1)
sumomark = ML("semisumo", ml_tavling)
core = Core("")
rumpa = Rumpa("")
axlar = Axlar("")
armar = Armar("")

# parent = ml, but intensity calculated against sumo's 1rm
smalmark = ML("smal", ml_tavling, reps={1 : (130, "2018-01-20")})
klotsving = Rumpa("klotsving", rumpa, related=ml_tavling)
enbensmark = Rumpa("enbensmark", rumpa, related=kb_tavling, time=2.6)

kb = KB("hög stång (utan bälte)", parent=kb_tavling, reps={1 : (85, "2017-11-11")}, time=3)
kb_pin = KB("pinböj (hög stång, utan bälte)", parent=kb_tavling, reps={1 : (70, "2017-11-11")}, time=3)
bredboj = KB("bredböj", parent=kb_tavling, reps={1 : (70, "2017-11-11")})
kb_fram = KB("fram (utan bälte)", parent=kb_tavling, reps={1 : (70, "2017-11-11")})
benspark = KB("benspark", related=kb_tavling, time=1.5)

negativa_rh = Core("Negativa RH", core, time=1.2)
latsdrag = Core("latsdrag", core, time=2)
raka_latsdrag = Core("raka latsdrag", core, time=2.1)
rygglyft = Core("rygglyft", core, related=ml_tavling, time=1.6)
sittande_rodd = Core("sittande rodd", core, time=1.5)

pinpress = BP("pinpress", bp_tavling, reps={1: (60, "2018-01-01")})
catapult = BP("Catapult", bp_tavling, reps={1: (90, "2018-01-01")})
bpfu = BP("FU, lill, stopp", bp_tavling, reps={1: (60, "2018-01-01")})

tricepsrep = Armar("tricepsrep", armar, related=bp_tavling, time=1.5)

militarpress = Axlar("Militärpress", axlar, reps={1: (35, "2017-10-10")}, time=3)

hangande_rodd = Axlar("hängande rodd", axlar, related=bp_tavling)
tryndrag = Axlar("tryndrag", axlar, related=bp_tavling, time=1.5)
"""

bp = BP("pekfinger, stopp", parent=bp_tavling, reps={1 : (65, "2018-01-20")}, time=5.1)
sumomark = ML("semisumo", ml_tavling)
core = Core("")
rumpa = Rumpa("")
axlar = Axlar("")
armar = Armar("")

# parent = ml, but intensity calculated against sumo's 1rm
smalmark = ML("smal", ml_tavling, reps={1 : (130, "2018-01-20")})
kronlyft = ML("kronlyft", ml_tavling, reps={1 : (130, "2018-01-20")})
rakmark = ML("raka", ml_tavling, reps={10 : (60, "2018-01-20")})
klotsving = Rumpa("klotsving", rumpa)
enbensmark = Rumpa("enbensmark", rumpa, time=2.6)

kb = KB("hög stång (utan bälte)", parent=kb_tavling, reps={1 : (85, "2017-11-11")}, time=3)
kb_pin = KB("pinböj (hög stång, utan bälte)", parent=kb_tavling, reps={1 : (70, "2017-11-11")}, time=3)
bredboj = KB("bredböj", parent=kb_tavling, reps={1 : (70, "2017-11-11")})
kb_fram = KB("fram (utan bälte)", parent=kb_tavling, reps={1 : (70, "2017-11-11")})
benspark = KB("benspark", time=1.5)

negativa_rh = Core("Negativa RH", core, time=1.2)
latsdrag = Core("latsdrag", core, time=2)
raka_latsdrag = Core("raka latsdrag", core, time=2.1)
rygglyft = Core("rygglyft", core, time=1.6)
sittande_rodd = Core("sittande rodd", core, time=1.5)

pinpress = BP("pinpress", bp_tavling, reps={1: (60, "2018-01-01")})
catapult = BP("Catapult", bp_tavling, reps={1: (90, "2018-01-01")})
bpfu = BP("FU, lill, stopp", bp_tavling, reps={1: (60, "2018-01-01")})

tricepsrep = Armar("tricepsrep", armar, time=1.5)

hantelflugor_omvanda = Axlar("omvända hantelflugor", axlar, time=1.5)
militarpress = Axlar("militärpress", axlar, reps={1: (35, "2017-10-10")}, time=3)

hangande_rodd = Axlar("hängande rodd", axlar)
tryndrag = Axlar("tryndrag", axlar, time=1.5)

###########################################################################################

File: js/johanna/bank.js
'use strict'; 
import * as E from '../lib/exercise.js'
/*
 * Exist only as root exercises. No work done on them.
 */
export var kb_tavling = new E.KB("", {'reps': new Map([[1, [110, "2017-11-11"]]])})
export var bp_tavling = new E.BP("", {'reps': new Map([[1, [65, "2018-02-24"]]])})
export var ml_tavling = new E.ML("", {'reps': new Map([[1, [130, "2018-01-20"]]])})
export var core = new E.Core("")
export var rumpa = new E.Rumpa("")
export var axlar = new E.Axlar("")
export var armar = new E.Armar("")

/*
 * Exercises actually used.
 */
export var bp = new E.BP("pekfinger, stopp", {'parent': bp_tavling, 'reps': new Map([[1, [65, "2017-11-11"]]]), 'time': 5.1})
export var sumomark = new E.ML("semisumo", {'parent': ml_tavling, 'reps': new Map([[1, [130, "2017-11-11"]]])})

// parent = new E.ml, but intensity calculated against sumo's 1rm
export var smalmark = new E.ML("smal", {'parent':ml_tavling, 'reps': new Map([[1, [130, "2018-01-20"]]])})
export var kronlyft = new E.ML("kronlyft", {'parent': ml_tavling, 'reps': new Map([[1, [130, "2018-01-20"]]])})
export var rakmark = new E.ML("raka", {'parent': ml_tavling}) // TODO: add a number? calculate 1RM from nRM? //, 'reps': new Map([[10, [160, "2018-01-20"]]])})
export var klotsving = new E.Rumpa("klotsving", {'parent': rumpa})
export var enbensmark = new E.Rumpa("enbensmark", {'parent': rumpa, 'time': 2.6})

export var kb = new E.KB("hög stång (utan bälte)", {'parent': kb_tavling, 'reps': new Map([[1, [85, "2017-11-11"]]]), 'time': 3})
export var kb_pin = new E.KB("pinböj (hög stång, utan bälte)", {'parent': kb_tavling, 'reps': new Map([[1 , [70, "2017-11-11"]]]), 'time': 3})
export var bredboj = new E.KB("bredböj", {'parent': kb_tavling, 'reps': new Map([[1 , [70, "2017-11-11"]]])})
export var kb_fram = new E.KB("fram (utan bälte)", {'parent': kb_tavling, 'reps': new Map([[1, [70, "2017-11-11"]]])})
export var benspark = new E.KB("benspark", {'time': 1.5})

export var negativa_rh = new E.Core("Negativa RH", {'parent': core, 'time': 1.2})
export var latsdrag = new E.Core("latsdrag", {'parent': core, 'time': 2})
export var raka_latsdrag = new E.Core("raka latsdrag", {'parent': core, 'time': 2.1})
export var rygglyft = new E.Core("rygglyft", {'parent': core, 'time': 1.6})
export var sittande_rodd = new E.Core("sittande rodd", {'parent': core, 'time': 1.5})

export var pinpress = new E.BP("pinpress", {'parent': bp_tavling, 'reps': new Map([[1, [60, "2017-11-11"]]])})
export var catapult = new E.BP("Catapult", {'parent': bp_tavling, 'reps': new Map([[1, [90, "2018-01-01"]]])})
export var bpfu = new E.BP("FU, lill, stopp", {'parent': bp_tavling, 'reps': new Map([[1, [60, "2018-01-01"]]])})

export var tricepsrep = new E.Armar("tricepsrep", {'parent': armar, 'time': 1.5})

export var hantelflugor_omvanda = new E.Axlar("omvända hantelflugor", {"parent": axlar, "time": 1.5})
export var militarpress = new E.Axlar("Militärpress", {'parent': axlar, 'reps': new Map([[1, [35, "2017-10-10"]]]), 'time':3})

export var hangande_rodd = new E.Axlar("hängande rodd", {'parent': axlar})
export var tryndrag = new E.Axlar("tryndrag", {'parent': axlar, 'time':1.5})

